vignettes/getting-osm-data.Rmd

```diff
@@ -0,0 +1,180 @@
+water <- osmdata_as_sf("natural", "water", bbox)
+waterbody <- bind_rows(water$osm_polygons, water$osm_multipolygons) |>
+  st_transform(epsg_code) |>
+  st_filter(waterway, .predicate = st_intersects) |>
+  st_union() |>
+  st_geometry()
+```
+
+## 3. Street network
+
+Querying the Overpass API for the `highway` key, using the expanded bounding box to include relevant streets close to the edge of the city:
+
+```{r}
+highways_value <- c("motorway", "trunk", "primary", "secondary", "tertiary")
+
+links_value <- sapply(X = highways_value,
+                      FUN = \(x) sprintf("%s_link", x),
+                      USE.NAMES = FALSE)
+
+highways <- osmdata_as_sf("highway",
+                          c(highways_value, links_value),
+                          bbox_expanded)
+```
+
+As some of elements are returend as polygons (see e.g. [this tutorial](https://geospatial-community.netlify.app/post/2022-03-31-spatial-networks/)) we cast those too into lines and combine them with the rest of the street network:
+
+```{r}
+# cast polygons (closed streets) into lines
+poly_to_lines <- highways$osm_polygons |>
+  st_cast("LINESTRING")
+
+# combine all features in one data frame
+highways_lines <- highways$osm_lines |>
+  bind_rows(poly_to_lines) |>
+  select("highway") |>  # only keep "highway" column
+  rename(type = `highway`) |>  # rename it to "type"
+  st_transform(epsg_code)
+```
+
+## 4. Rail network
+
+Querying the Overpass API for the `railway:rail` key:tag, also using the expanded bounding box to include relevant ways close to the edge of the city:
+
+```{r}
+railways <- osmdata_as_sf("railway", "rail", bbox_expanded)
+railways_lines <- railways$osm_lines |>
+  select("railway") |>  # only keep "railway" column
+  rename(type = `railway`) |>  # rename it to "type"
+  st_transform(epsg_code)
+```
+
+## Write OSM data to file
+
+```{r}
+st_write_list <- list(
+  city_boundary = bucharest$boundary,
+  waterway = bucharest$river_centerline,
+  waterbody = bucharest$river_surface,
+  highways = bucharest$streets,
+  railways = railways_lines
+)
+
+purrr::walk(
+  st_write_list,
+  ~ st_write(
+    .x,
+    dsn = sprintf("%s_%s.gpkg",
+                  names(st_write_list)[purrr::map_lgl(st_write_list,
+                                                      identical,
+                                                      .x)], city_name),
+    append = FALSE,
+    quiet = TRUE
+  )
+)
+```
+
+
+## Visualise OSM data
+
+```{r, eval=TRUE, fig.alt="All layers combined", fig.cap="All layers combined"}
+if (requireNamespace("ggplot2", quietly = TRUE)) {
+  library(ggplot2)
+  ggplot() +
+    geom_sf(data = city_boundary, fill = "grey", color = "black") +
+    geom_sf(data = railways_lines, color = "orange") +
+    geom_sf(data = highways_lines, color = "black") +
+    geom_sf(data = waterbody, fill = "blue", color = "blue") +
+    geom_sf(data = waterway, color = "blue")
+} else {
+  message("ggplot2 not available; skipping plot examples.")
+}
```

Comment:
As `ggplot2` is in the `Suggests` field of `DESCRIPTION`, it might not be installed on the user's system. In that case, when built locally, the vignette will display a message instead of the plot. This approach ensures that the pacakge passes `R CMD check` and the vignette is built even though `ggplot2` is not used anywhere else in the package.

Reply:
Nice solution!